api/preprints/serializers.py

```diff
@@ -81,6 +82,7 @@ class PreprintSerializer(JSONAPISerializer):
     description = ser.CharField(required=False, allow_blank=True, allow_null=True, source='node.description')
     tags = JSONAPIListField(child=NodeTagField(), required=False, source='node.tags')
     node_is_public = ser.BooleanField(read_only=True, source='node__is_public')
+    preprint_doi_on_datacite = ser.SerializerMethodField(read_only=True)
```

Comment:
For performance, I think this should only be returned in the preprint detail endpoint. Otherwise, we'd be adding at least one query per preprint on the preprint list endpoint.

I'll confirm with @jamescdavis that it's only needed on the detail endpoint.

Reply:
OK, discussed with James. He pointed that for consistency, it would be better to return the flag in both the list and the detail endpoint.

To avoid having to do n queries in the list endpoint, I recommend

- Store the datetime the DOI was created in a new field `preprint_doi_created`. Make the value `null` if the DOI hasn't been created
- Expose that datetime in this serializer